rpc/chain/client.go

```diff
@@ -0,0 +1,678 @@
+package chain
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/afex/hystrix-go/hystrix"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/status-im/status-go/services/rpcstats"
+)
+
+type FeeHistory struct {
+	BaseFeePerGas []string `json:"baseFeePerGas"`
+}
+
+type ClientWithFallback struct {
+	ChainID  uint64
+	main     *ethclient.Client
+	fallback *ethclient.Client
+
+	mainRpc     *rpc.Client
+	fallbackRpc *rpc.Client
+}
+
+func NewSimpleClient(main *rpc.Client, chainID uint64) *ClientWithFallback {
+	hystrix.ConfigureCommand(fmt.Sprintf("ethClient_%d", chainID), hystrix.CommandConfig{
+		Timeout:               10000,
+		MaxConcurrentRequests: 100,
+		SleepWindow:           300000,
+		ErrorPercentThreshold: 25,
+	})
+
+	return &ClientWithFallback{
+		ChainID:     chainID,
+		main:        ethclient.NewClient(main),
+		fallback:    nil,
+		mainRpc:     main,
+		fallbackRpc: main,
+	}
+}
+
+func NewClient(main, fallback *rpc.Client, chainID uint64) *ClientWithFallback {
+	hystrix.ConfigureCommand(fmt.Sprintf("ethClient_%d", chainID), hystrix.CommandConfig{
+		Timeout:               10000,
+		MaxConcurrentRequests: 100,
+		SleepWindow:           300000,
+		ErrorPercentThreshold: 25,
+	})
+
+	var fallbackEthClient *ethclient.Client
+	if fallback != nil {
+		fallbackEthClient = ethclient.NewClient(fallback)
+	}
+	return &ClientWithFallback{
+		ChainID:     chainID,
+		main:        ethclient.NewClient(main),
+		fallback:    fallbackEthClient,
+		mainRpc:     main,
+		fallbackRpc: fallback,
+	}
+}
+
+func (c *ClientWithFallback) Close() {
+	c.main.Close()
+	c.fallback.Close()
+}
+
+func (c *ClientWithFallback) makeCallNoReturn(main func() error, fallback func() error) error {
+	output := make(chan bool, 1)
```

Comment:
```suggestion
	output := make(chan struct{}, 1)
```

Reply:
If the output value is not used, then a signal channel can be used (these don't use memory at all and are cooler ;) )